File: src/app/Domain/HeaderType.php
<?php

declare(strict_types=1);

namespace Domain\HeaderType;

enum HeaderType: string
{
    case Json = 'application/json';
    case PlainText = 'text/plain';

    public function header(): string
    {
        return "Content-Type: {$this->value}; charset=utf-8";
    }
}

File: tests/header_type.py
from enum import Enum


class HeaderType(str, Enum):
    """
    Enum for common content types used in HTTP responses.
    """

    HTML_TEXT = "text/html"
    PLAIN_TEXT = "text/plain"

    def __init__(self, content_type: str) -> None:
        self.content_type = f"{content_type}; charset=UTF-8"
